ua/org.eclipse.tips.core/src/org/eclipse/tips/core/TipImage.java

```diff
@@ -98,7 +97,6 @@ public TipImage(String base64Image) {
 			int from = base64Image.indexOf('/') + 1;
 			int to = base64Image.indexOf(';');
 			setExtension(base64Image.substring(from, to).trim());
```

Comment:
Please replace the `setExtension` call with a direct assignment to `fExtension`. First, this avoids the unnecessary replacement in the base 64 representation of the image. Second, calling non-final, non-private methods in a constructor can be dangerous, as a subclass may override them.

Reply:
Thank your for addressing this, I think this change is useful. I included this advice in the commit and also updated the commit message to focus on the behaviour change. Please merge the PR if you think that it is more clear now.